Observer/App/Action/Action.php

```diff
@@ -0,0 +1,82 @@
+<?php
+/**
+ * Copyright (c) 2019, Nosto Solutions Ltd
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without modification,
+ * are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its contributors
+ * may be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * @author Nosto Solutions Ltd <contact@nosto.com>
+ * @copyright 2019 Nosto Solutions Ltd
+ * @license http://opensource.org/licenses/BSD-3-Clause BSD 3-Clause
+ *
+ */
+
+namespace Nosto\Cmp\Observer\App\Action;
+
+use Magento\Framework\Event\Observer;
+use Magento\Framework\Event\ObserverInterface;
+use Magento\Framework\App\Response\Http as HttpResponse;
+use Magento\Framework\App\Request\Http as HttpRequest;
+use Nosto\Cmp\Helper\TimeHelper;
+
+class Action implements ObserverInterface
+{
+    // app/code/Magento/Catalog/etc/frontend/page_types.xml
+    const CATALOG_PAGE_TYPE = 'catalog_category_view';
+
+    /**
+     * @var HttpResponse $response
+     */
+    private $response;
+    /**
+     * @var HttpRequest
+     */
+    private $request;
+
+    /**
+     * Action constructor.
+     * @param HttpResponse $response
+     * @param HttpRequest $request
+     */
+    public function __construct(
+        HttpResponse $response,
+        HttpRequest $request
+    )
+    {
+        $this->response = $response;
+        $this->request = $request;
+    }
+    public function execute(Observer $observer)
+    {
+        if ((string)$this->request->getFullActionName() === self::CATALOG_PAGE_TYPE) {
```

Comment:
Do we need to check the page type? I would just check that if there are elements (we need a checker method in `TimeHelper`) in `TimeHelper` and if there are, add it to the header